src/compute-client/src/controller.rs

```diff
@@ -278,6 +281,10 @@ pub struct ComputeController<T> {
     envd_epoch: NonZeroI64,
     replica_metrics_streams:
         BTreeMap<ReplicaId, BoxStream<'static, Result<Vec<ServiceProcessMetrics>, anyhow::Error>>>,
+    /// Periodic notification to produce a `ReplicaWriteFrontiers` response.
+    stats_update_ticker: tokio::time::Interval,
+    /// Set to `true` if `process` should produce a `ReplicaWriteFrontiers` next.
+    stats_update_pending: bool,
```

Comment:
Instead of recording the frontiers on a timer, can we just update it whenever we handle a `FrontiersUpper` message ?

(The reason we use a timer for the metrics table is because there is no way other than polling to be notified of changes to the values from `metrics-server`.)

Reply:
Can you do some analysis on how often `FrontierUppers` can arrive, and ask in #team-storage whether we should be concerned about that frequency of updates?

We should only do the more complicated solution with the timer if it's actually necessary, IMO